marketplace/plugins/supabase/lib/index.ts

```diff
@@ -0,0 +1,207 @@
+    return updateQueryRes;
+  }
+
+  async deleteRows(queryOptions: QueryOptions, supabaseClient: any) {
+    const { delete_table_name, delete_filters, delete_sort, delete_limit } = queryOptions;
+    let deleteQuery: any = supabaseClient.from(delete_table_name).delete();
+    if (delete_filters) {
+      const deleteFiltersData: Filter[] = Object.values(delete_filters);
+      this.addQueryFilters(deleteQuery, deleteFiltersData);
+    }
+    if (delete_sort) {
+      const deleteSorts: Sort[] = Object.values(delete_sort);
+      this.addQuerySort(deleteQuery, deleteSorts);
+    }
+    if (delete_limit) {
+      deleteQuery = deleteQuery.limit(Number(delete_limit));
+    }
+    const res: any = await deleteQuery;
+    return res;
+  }
+
+  async countRows(queryOptions: QueryOptions, supabaseClient: any) {
+    const { count_table_name, count_filters } = queryOptions;
+    const countQuery: any = supabaseClient.from(count_table_name).select('count', { count: 'exact' });
+    if (count_filters) {
+      const countFiltersData: Filter[] = Object.values(count_filters);
+      this.addQueryFilters(countQuery, countFiltersData);
+    }
+    const res: any = await countQuery;
+    return res;
+  }
+
+  addQueryFilters(query: any, filters: Filter[]) {
+    filters.forEach((filter: any) => {
+      const { operator, column, value } = filter;
+      if (operator === '==') {
+        query = query.eq(column, value);
+      } else if (operator === '!=') {
+        query = query.neq(column, value);
+      } else if (operator === '<') {
+        query = query.lt(column, value);
+      } else if (operator === '>') {
+        query = query.gt(column, value);
+      } else if (operator === '<=') {
+        query = query.lte(column, value);
+      } else if (operator === '>=') {
+        query = query.gte(column, value);
+      } else if (operator === 'is') {
+        query = query.is(column, value);
+      } else if (operator === 'in') {
+        query = query.in(column, JSON.parse(value));
+      } else if (operator === 'is not') {
+        query = query.not(column, 'is', value);
+      }
+    });
+  }
+
+  addQuerySort(query: any, sorts: Sort[]) {
+    sorts.forEach((sort: any) => {
+      const { column, order } = sort;
+      query = query.order(column, { ascending: order === 'ascend' });
+    });
+  }
+
+  async getConnection(sourceOptions: SourceOptions, _options?: object): Promise<any> {
+    const { project_url, service_role_secret } = sourceOptions;
+    // Create a single supabase client for interacting with your database
+    const supabaseClient = createClient(project_url, service_role_secret);
+
+    return supabaseClient;
+  }
+
+  async testConnection(sourceOptions: SourceOptions): Promise<ConnectionTestResult> {
+    const supabaseClient = await this.getConnection(sourceOptions);
+
+    if (!supabaseClient) {
+      throw new Error('Invalid credentials');
+    }
+
+    return {
+      status: 'ok',
+    };
+  }
```

Comment:
@blank0537 while testing we found that client auth is only done when we run queries. There isn't no officially supported method which we can use for testing session with anon keys.

From fiddling with it, I found we can do something like this instead which would force auth first.

```js
  async testConnection(sourceOptions: SourceOptions): Promise<ConnectionTestResult> {
    const supabaseClient = await this.getConnection(sourceOptions);

    try {
      const res = await supabaseClient.from('').select('1');

      if (res.error) {
        throw new QueryError(`Connection test failed`, res.error, {});
      }

      return {
        status: 'ok',
      };
    } catch (error) {
      throw new QueryError('Connection test failed', error.message, {});
    }
  }
}
```

Could you help make the change?

Reply:
@akshaysasidrn this is done